src/main/java/com/github/holly/accountability/password_reset_email/PasswordEmailController.java

```diff
@@ -0,0 +1,152 @@
+    @ResponseBody
+    @GetMapping("/send-token")
+    public GenericResponse sendPasswordEmail(@RequestParam String email) {
+        try {
+            return passwordEmailService.sendPasswordEmail(email);
+        } catch (Exception e) {
+            return new GenericResponse(e.getMessage());
+        }
+    }
+
+    @GetMapping(CHANGE_PASSWORD_FROM_TOKEN + "/{token}")
+    public ResponseEntity<?>  showChangePasswordFromTokenPage(@PathVariable("token") String token) {
+        boolean isValid = passwordEmailService.validatePasswordResetToken(token);
+
+        if (!isValid) {
+
+            String loginUrl = UriComponentsBuilder
+                    .fromUriString(applicationProperties.getBaseUrl())
+                    .path("/login")
+                    .build()
+                    .encode(StandardCharsets.UTF_8)
+                    .toUriString();
+
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .header("Location", loginUrl)
+                    .build();
+        }
+
+        String tokenPasswordChangeUrl = UriComponentsBuilder
+                .fromUriString(applicationProperties.getBaseUrl())
+                .path(CHANGE_PASSWORD_FROM_TOKEN)
+                .queryParam("token", token)
+                .build()
+                .encode(StandardCharsets.UTF_8)
+                .toUriString();
+
+        return ResponseEntity.status(HttpStatus.FOUND)
+                .header("Location", tokenPasswordChangeUrl)
+                .build();
+    }
+
+    //change this so it's a JSON object being given with a PostMapping
+    @ResponseBody
+    @PostMapping("/set-new-password")
+    public ResponseEntity<?> changePasswordFromToken(@RequestBody @Valid ResetPasswordDto passwordDto,
+                                                     BindingResult bindingResult) {
+
+        if (bindingResult.hasErrors()) {
+            Map<String, String> errorMap = new HashMap<>();
+
+            for (FieldError error : bindingResult.getFieldErrors()) {
+                errorMap.put(error.getField(), error.getDefaultMessage());
+            }
+
+            return ResponseEntity.badRequest()
+                    .body(new ErrorResponse("Validation failed", errorMap));
+        }
+
+        try {
+            ResetPasswordDto newPassword = passwordEmailService.setNewPassword(passwordDto.getToken(), passwordDto);
+            return ResponseEntity
+                    .ok(new SuccessfulPasswordChangeResponse(
+                            "Password changed successfully!", newPassword));
+
+        } catch (PasswordEmailService.PasswordsNotMatchingException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                    .body(new ErrorResponse("Passwords do not match"));
+
+        } catch (Exception e) {
+            return ResponseEntity.internalServerError()
+                    .body(new ErrorResponse("An unexpected error occurred"));
+        }
+    }
+
+    public class ErrorResponse {
```

Comment:
This class doesn't need to be an inner class. It can be used in a lot of places in your application